ggipcd/src/ipc_components.c

```diff
@@ -67,11 +75,101 @@ static void set_component_name(
     component_name_lengths[handle - 1] = (uint8_t) component_name.len;
 }
 
+static GglError verify_svcuid(void *ctx, GglMap params, uint32_t handle) {
+    (void) ctx;
+    (void) handle;
+
+    GglObject *svcuid_buf;
+
+    GglError ret = ggl_map_validate(
+        params,
+        GGL_MAP_SCHEMA({ GGL_STR("svcuid"), true, GGL_TYPE_BUF, &svcuid_buf }, )
+    );
+
+    if (ret != GGL_ERR_OK) {
+        GGL_LOGE("Failed to validate the map provided to 'verify_svcuid'.");
+        return GGL_ERR_INVALID;
+    }
+
+    if (svcuid_buf->type != GGL_TYPE_BUF) {
+        GGL_LOGE("svcuid must be of type buffer.");
+        return GGL_ERR_INVALID;
+    }
+
+    GglBuffer svcuid_bin = GGL_BUF((uint8_t[SVCUID_BIN_LEN]) { 0 });
+    bool decoded = ggl_base64_decode(svcuid_buf->buf, &svcuid_bin);
+    if (!decoded) {
+        GGL_LOGE("svcuid is invalid base64.");
+        return GGL_ERR_INVALID;
+    }
+
+    GGL_MTX_SCOPE_GUARD(&ggl_ipc_component_registered_components_mtx);
+
+    for (GglComponentHandle i = 1; i <= registered_components; i++) {
+        GglBuffer svcuid_bin_i = GGL_BUF(svcuids[i - 1]);
+        if (ggl_buffer_eq(svcuid_bin, svcuid_bin_i)) {
+            GGL_LOGD("Found the requested svcuid.");
+            ggl_respond(handle, GGL_OBJ_BOOL(true));
+            return GGL_ERR_OK;
+        }
+    }
+
+    GGL_LOGE("Requested svcuid not found.");
+
+    ggl_respond(handle, GGL_OBJ_BOOL(false));
+    return GGL_ERR_OK;
+}
+
+static void *ggl_ipc_component_server(void *args) {
+    (void) args;
+
+    // Hack. Fix it. Need IPC to be up and running before we start this server.
```

Comment:
Adding `TODO` might make this easier to find later

Reply:
is it even needed? it shouldnt depend on the other thread

Reply:
removed.